spec/app/ics-028-cross-chain-validation/technical_specification.md

```diff
@@ -531,10 +550,23 @@ function InitGenesis(state: ConsumerGenesisState) {
 - Expected postcondition:
   - The capability for the port `ConsumerPortId` is claimed.
   - A client of the provider chain is created and the client ID is stored into `providerClient`.
+  - The initial validator set is returned to the consensus engine.
 - Error condition:
+  - The genesis state contains no valid provider client state, where the validity is defined as in [ICS 7](../../client/ics-007-tendermint-client).
+  - The genesis state contains no valid provider consensus state, where the validity is defined as in [ICS 7](../../client/ics-007-tendermint-client).
+  - The genesis state contains an empty initial validator set.
+  - The genesis state contains an initial validator set that does not match the validator set in the provider consensus state.
   - The capability for the port `ConsumerPortId` cannot be claimed.
 
-> **Note**: CCV assumes that `ConsumerGenesisState` is disseminated to all validators. To ensure that validators receive the correct genesis state, the governance proposal also contains a hash of the `ConsumerGenesisState` (see `CreateConsumerChainProposal`). 
+> **Note**: CCV assumes that the _same_ consumer chain genesis state is disseminated to all the correct validators in the initial validator set of the consumer chain. 
```

Comment:
"all the correct validators": Does that mean to "all the operators managing keys of validators?". Perhaps this should be more explained in what the operators have to do to launch the chain. Where to get information, and where to put it, etc?

Reply:
I agree that should be clear for operators, but I'm not sure it should be part of the spec. Probably is better to have a tutorial or FAQ for operators on how to deal w/ interchain security. What do you think?